packages/gatsby-transformer-documentationjs/src/gatsby-node.js

```diff
@@ -80,76 +60,166 @@ exports.onCreateNode = async ({
   }
 
   if (documentationJson && documentationJson.length > 0) {
-    documentationJson.forEach((docsJson, i) => {
-      const picked = _.pick(docsJson, [`kind`, `memberof`, `name`, `scope`])
-
-      // Defaults
-      picked.params = [{ name: ``, type: { type: ``, name: `` } }]
-      picked.returns = [{ type: { type: ``, name: `` } }]
-      picked.examples = [{ raw: ``, highlighted: `` }]
-
-      // Prepare various sub-pieces.
-      if (docsJson.description) {
-        picked.description___NODE = createDescriptionNode(
-          node,
-          commentId(node.id, i),
-          stringifyMarkdownAST(docsJson.description),
-          `comment.description`,
-          actions,
-          createNodeId
-        )
+    const handledDocs = new WeakMap()
+    const typeDefs = new Map()
+
+    const getNodeIDForType = typeName => {
+      if (typeDefs.has(typeName)) {
+        return typeDefs.get(typeName)
       }
 
-      const transformParam = param => {
-        if (param.description) {
-          param.description___NODE = createDescriptionNode(
-            node,
-            commentId(node.id, i),
-            stringifyMarkdownAST(param.description),
-            param.name,
-            actions,
-            createNodeId
-          )
-          delete param.description
-        }
-        delete param.lineNumber
-
-        // When documenting destructured parameters, the name
-        // is parent.child where we just want the child.
-        if (param.name.split(`.`).length > 1) {
-          param.name = param.name
-            .split(`.`)
-            .slice(-1)
-            .join(`.`)
-        }
+      const index = documentationJson.findIndex(
```

Comment:
You're only using the index to use it as `documentationJson[index]` so might be better to use find instead. Unless i've missed something
```suggestion
      const comment = documentationJson.find(
```

Reply:
I also use it as `commentNumber` field (passing it in line 79) - this is added to node (for backward compatibility)